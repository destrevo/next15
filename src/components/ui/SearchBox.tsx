"use client";
import React, { JSX } from "react";

/**
 * SearchBox component that renders an input field for searching projects.
 *
 * @returns {JSX.Element} The rendered SearchBox component.
 */
const SearchBox: React.FC = (): JSX.Element => {
  return (
    <input
      className="z-10 h-full w-[270px] mr-2.5 flex items-center cursor-pointer self-baseline font-poppins font-normal text-base leading-4 mb-2.5 md:mb-0 bg-transparent border-none outline-none focus:ring-0 placeholder-white"
      placeholder="Search for Project"
    />
  );
};

export default SearchBox;
