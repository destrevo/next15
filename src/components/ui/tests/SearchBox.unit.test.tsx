import React from "react";
import { render } from "@testing-library/react";
import SearchBox from "../SearchBox";

describe("SearchBox", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<SearchBox />);
    expect(asFragment()).toMatchSnapshot();
  });
});
