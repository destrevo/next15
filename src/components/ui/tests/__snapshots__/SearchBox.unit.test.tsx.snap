// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<DocumentFragment>
  <input
    class="z-10 h-full w-[270px] mr-2.5 flex items-center cursor-pointer self-baseline font-poppins font-normal text-base leading-4 mb-2.5 md:mb-0 bg-transparent border-none outline-none focus:ring-0 placeholder-white"
    placeholder="Search for Project"
  />
</DocumentFragment>
`;
